src/gui/search/searchjobwidget.cpp

```diff
@@ -280,22 +280,23 @@ void SearchJobWidget::downloadTorrent(const QModelIndex &rowIndex)
 
     if (torrentUrl.startsWith("magnet:", Qt::CaseInsensitive))
     {
-        addTorrentToSession(torrentUrl);
+        addTorrentToSession(torrentUrl, showDialog);
     }
     else
     {
         SearchDownloadHandler *downloadHandler = m_searchHandler->manager()->downloadTorrent(siteUrl, torrentUrl);
-        connect(downloadHandler, &SearchDownloadHandler::downloadFinished, this, &SearchJobWidget::addTorrentToSession);
+        connect(downloadHandler, &SearchDownloadHandler::downloadFinished
+            , this, [this, showDialog](const QString &source) { addTorrentToSession(source, showDialog); });
         connect(downloadHandler, &SearchDownloadHandler::downloadFinished, downloadHandler, &SearchDownloadHandler::deleteLater);
     }
     setRowColor(rowIndex.row(), QApplication::palette().color(QPalette::LinkVisited));
 }
 
-void SearchJobWidget::addTorrentToSession(const QString &source)
+void SearchJobWidget::addTorrentToSession(const QString &source, ShowAddNewTorrentDialog showDialog)
 {
     if (source.isEmpty()) return;
 
-    if (AddNewTorrentDialog::isEnabled())
+    if (showDialog == ShowAddNewTorrentDialog::Yes || (showDialog == ShowAddNewTorrentDialog::FromSettings && AddNewTorrentDialog::isEnabled()))
```

Comment:
coding guidelines require parentheses to each comparison:
`if ((showDialog == ShowAddNewTorrentDialog::Yes) || ((showDialog == ShowAddNewTorrentDialog::FromSettings) && AddNewTorrentDialog::isEnabled()))`

Reply:
Fixed.